packages/core/src/wallet/IndyWallet.ts

```diff
@@ -279,95 +264,85 @@ export class IndyWallet implements Wallet {
-    recipientKeys: Verkey[],
-    senderVk: Verkey
+    recipientKeys: string[],
+    senderVerkey?: string | null
   ): Promise<PackedMessage> {
-    const messageRaw = JsonEncoder.toBuffer(payload)
-    const packedMessage = await this.indy.packMessage(this.walletHandle, messageRaw, recipientKeys, senderVk)
-    return JsonEncoder.fromBuffer(packedMessage)
+    try {
+      const messageRaw = JsonEncoder.toBuffer(payload)
+      const packedMessage = await this.indy.packMessage(
+        this.walletHandle,
+        messageRaw,
+        recipientKeys,
+        senderVerkey ?? null
+      )
+      return JsonEncoder.fromBuffer(packedMessage)
+    } catch (error) {
+      throw new WalletError('Error packing message', { cause: error })
+    }
   }
 
   public async unpack(messagePackage: PackedMessage): Promise<UnpackedMessageContext> {
-    const unpackedMessageBuffer = await this.indy.unpackMessage(this.walletHandle, JsonEncoder.toBuffer(messagePackage))
-    const unpackedMessage = JsonEncoder.fromBuffer(unpackedMessageBuffer)
-    return {
-      ...unpackedMessage,
-      message: JsonEncoder.fromString(unpackedMessage.message),
+    try {
+      const unpackedMessageBuffer = await this.indy.unpackMessage(
+        this.walletHandle,
+        JsonEncoder.toBuffer(messagePackage)
+      )
+      const unpackedMessage = JsonEncoder.fromBuffer(unpackedMessageBuffer)
+      return {
+        senderVerkey: unpackedMessage.sender_verkey,
+        recipientVerkey: unpackedMessage.recipient_verkey,
+        message: JsonEncoder.fromString(unpackedMessage.message),
+      }
+    } catch (error) {
+      throw new WalletError('Error unpacking message', { cause: error })
     }
   }
 
-  public async sign(data: Buffer, verkey: Verkey): Promise<Buffer> {
-    const signatureBuffer = await this.indy.cryptoSign(this.walletHandle, verkey, data)
-
-    return signatureBuffer
-  }
-
-  public async verify(signerVerkey: Verkey, data: Buffer, signature: Buffer): Promise<boolean> {
-    // check signature
-    const isValid = await this.indy.cryptoVerify(signerVerkey, data, signature)
-
-    return isValid
-  }
-
-  public async addWalletRecord(type: string, id: string, value: string, tags: Record<string, string>) {
-    return this.indy.addWalletRecord(this.walletHandle, type, id, value, tags)
-  }
-
-  public async updateWalletRecordValue(type: string, id: string, value: string) {
-    return this.indy.updateWalletRecordValue(this.walletHandle, type, id, value)
-  }
-
-  public async updateWalletRecordTags(type: string, id: string, tags: Record<string, string>) {
-    return this.indy.addWalletRecordTags(this.walletHandle, type, id, tags)
-  }
-
-  public async deleteWalletRecord(type: string, id: string) {
-    return this.indy.deleteWalletRecord(this.walletHandle, type, id)
-  }
+  public async sign(data: Buffer, verkey: string): Promise<Buffer> {
+    try {
+      const signatureBuffer = await this.indy.cryptoSign(this.walletHandle, verkey, data)
```

Comment:
Very nitpicky, but I prefer to just instantly return the value instead of saving it in a variable.